bokehjs/src/lib/core/visuals.ts

```diff
@@ -1,14 +1,21 @@
 import * as mixins from "./property_mixins"
 import * as p from "./properties"
-import {color2css} from "./util/color"
+import {color2rgba, decode_rgba} from "./util/color"
 import {Context2d} from "./util/canvas"
 import {Class} from "./class"
-import {Arrayable} from "./types"
-import {map} from "./util/arrayable"
+import {Arrayable, Color} from "./types"
+import {isString} from "./util/types"
+import {subselect} from "./util/arrayable"
 import {LineJoin, LineCap, FontStyle, TextAlign, TextBaseline} from "./enums"
 
 import {HasProps} from "./has_props"
 import {ColumnarDataSource} from "models/sources/columnar_data_source"
+import {Texture} from "models/textures/texture"
+
+function color2css(color: Color | number, alpha: number): string {
+  const [r, g, b, a] = isString(color) ? color2rgba(color) : decode_rgba(color)
+  return `rgba(${r*255}, ${g*255}, ${b*255}, ${a == 1.0 ? alpha : a})`
+}
```

Comment:
why not keep this in `util/color` together with the other very similar helpers in one place?

Reply:
This is a special case just for visuals, so I keep it here, at least for now. Based on this work, I need to rethink color manipulation once more, and then I will likely remove this and handle colors properly consistently across bokehjs.